modules/irsa/main.tf

```diff
@@ -28,7 +44,7 @@ resource "kubernetes_service_account_v1" "irsa" {
     }
   }
 
-  automount_service_account_token = true
+  automount_service_account_token = false
```

Comment:
If you set it to false, kubelet won't mount the credentials into the pod automatically causing the pod to fail.